apps/cowswap-frontend/src/legacy/state/orders/reducer.ts

```diff
@@ -347,27 +346,40 @@ export default createReducer(initialState, (builder) =>
     })
     .addCase(fulfillOrdersBatch, (state, action) => {
       prefillState(state, action)
-      const { ordersData, chainId, isSafeWallet } = action.payload
+      const { orders, chainId, isSafeWallet } = action.payload
 
       // if there are any newly fulfilled orders
       // update them
-      ordersData.forEach(({ id, fulfillmentTime, transactionHash, apiAdditionalInfo }) => {
-        const orderObject = getOrderById(state, chainId, id)
+      orders.forEach((order) => {
+        const { uid, ethflowData } = order
+        const orderObject = getOrderById(state, chainId, uid)
 
         if (orderObject) {
-          deleteOrderById(state, chainId, id)
+          deleteOrderById(state, chainId, uid)
 
           orderObject.order.status = OrderStatus.FULFILLED
-          orderObject.order.fulfillmentTime = fulfillmentTime
+          orderObject.order.fulfillmentTime = new Date().toISOString()
 
-          orderObject.order.fulfilledTransactionHash = transactionHash
           orderObject.order.isCancelling = false
           // EthFlow orders validTo is different
-          orderObject.order.validTo = getValidTo(apiAdditionalInfo, orderObject.order)
-
-          orderObject.order.apiAdditionalInfo = apiAdditionalInfo
+          orderObject.order.validTo = getValidTo(ethflowData?.userValidTo, orderObject.order)
+
+          orderObject.order.apiAdditionalInfo = {
+            creationDate: order.creationDate,
+            availableBalance: order.availableBalance,
+            executedBuyAmount: order.executedBuyAmount,
+            executedSellAmount: order.executedSellAmount,
+            executedSellAmountBeforeFees: order.executedSellAmountBeforeFees,
+            executedFeeAmount: order.executedFeeAmount,
+            executedSurplusFee: order.executedSurplusFee,
+            invalidated: order.invalidated,
+            ethflowData: order.ethflowData,
+            onchainOrderData: order.onchainOrderData,
+            class: order.class,
+            fullAppData: order.fullAppData,
+          }
```

Comment:
Why manually recreate apiAdditionalInfo?

It was meant as a quick way to add api info not part of the order modal.
if that's modelled elsewhere, we could get rid of it.

Reply:
It used to be a part of `fetchOrderPopupData`.
But I refactored it in order to make it more generic via using `EnrichedOrder`.
So, mainly, the code was migrated from `fetchOrderPopupData` to the reducer.

<img width="1215" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/f3d3e0cb-d7a6-47ac-9fd0-c84d05a2798b">